Source/WebCore/dom/ShadowRoot.h

```diff
@@ -181,6 +186,8 @@ class ShadowRoot final : public DocumentFragment, public TreeScope {
     std::unique_ptr<Style::Scope> m_styleScope;
     std::unique_ptr<SlotAssignment> m_slotAssignment;
     mutable std::optional<PartMappings> m_partMappings;
+
+    AtomString m_referenceTarget;
```

Comment:
This is potentially controversial to add to every shadow root for a feature likely to be relatively seldom used. Is there any good alternative?

Reply:
We could consider adding it to `NodeRareData` but I doubt that one extra pointer will matter in the grand scheme of things.